### PR TITLE
Fix wrong formula order

### PR DESCRIPTION
**Pull request by rikhuijzer**
- Fixes https://github.com/JuliaAI/MLJGLMInterface.jl/issues/34

### CODE REVIEW

**Comment by ablaom** [comment]
Thanks for sorting this @rikhuijzer!

Just to confirm, the fix is backwards compatible, right? That is, there was no need to bump the compat requirement for GLM?

**Comment by rikhuijzer** [comment]
> Just to confirm, the fix is backwards compatible, right? That is, there was no need to bump the compat requirement for GLM?

Yes correct. `MLJGLMInterface` is more reliable now so things should be fine irregardless also of whether the bug is fixed upstream or not. Also the overhead of the workaround/patch shouldn't be too bad

File: src/MLJGLMInterface.jl [hunk 1]
```diff
@@ -291,17 +291,29 @@ function glm_report(glm_model, features, reportkeys)
...
 
 Return formula which is ready to be passed to `fit(form, data, ...)`.
 """
-function glm_formula(model, features)::FormulaTerm
+function glm_formula(model, features::AbstractVector{Symbol})::FormulaTerm
     # By default, using a JuliaStats formula will add an intercept.
     # Adding a zero term explicitly disables the intercept.
     # See the StatsModels.jl tests for more information.
     intercept_term = model.fit_intercept ? 1 : 0
     form = GLM.Term(:y) ~ sum(GLM.term.(features)) + GLM.term(intercept_term)
```
Comment (kleinschmidt) [inline src/MLJGLMInterface.jl:303]: instead you can do

```julia
form = GLM.StatsModels.FormulaTerm(GLM.Term(:y), sum(GLM.term.(features)) + GLM.term(intercept_term)))
```

then you don't need to do the cleanup.

the sorting of terms is only triggered by the `~` operator, not the constructor:

```julia
julia> term(:y) ~ term(:a) + term(:b) + term(1)
FormulaTerm
Response:
  y(unknown)
Predictors:
  1
  a(unknown)
  b(unknown)

julia> FormulaTerm(term(:y), term(:a) + term(:b) + term(1))
FormulaTerm
Response:
  y(unknown)
Predictors:
  a(unknown)
  b(unknown)
  1
```

**Review by kleinschmidt — COMMENTED** [review]
given that this is relying pretty heavily on StatsModels stuff (not just GLM), I'd also recommend taking a dep on StatsModels itself as well and specifying compatibility bounds so that you have to opt into this kind of breakage.

**Pull request merged**